torch/fx/experimental/symbolic_shapes.py

```diff
@@ -3213,8 +3230,8 @@ def _nested_int_aware_sort(tup):
                 stride[i] = self.create_symbol(
```

Comment:
There's something a bit funny going on here.

Naively, if I have something like: `size = (2, 3), stride = (3, 1)` and I specify `size = (STATIC, STATIC), stride = (DYNAMIC, STATIC)`, I would expect to end up with symbolic variable allocation `size = (s0, s1), stride = (s2, 1)`; notably, s2 != s1 and I allow for `stride[0]` to vary even though everything else is constant.

However, per the logic here, the DYNAMIC entry here would actually not do anything. This is because above we would find that the stride 3 it will be present in candidates, and can be directly allocated as s1, bypassing the usage of `dynamic_dim` entirely.

This seems pretty counterintuitive to me!

It seems worth getting this right here, because it affects how downstream callers to strides go. Specifically, I feel we need a new designation like INFER, which says that you should preferentially try to infer the stride from sizes, before allocating a new stride (it's not clear to me if you should then decay to DUCK or not, I guess follow whatever we used to do.)

Reply:
For the example, why would you want the size = (s0, s1). I agree with DYNAMIC being ignored because of candidates, but dont understand why sizes should be dynamic too.

Reply:
I am going to send a commit which does following thing 
~~~

x = torch.arange(18).reshape((3, 6))
print(x)
print(x.shape, x.stride())
print("-----")

x1 = x[:,:3]
print(x1)
print(x1.shape, x1.stride())
print("-----")

x2 = torch.arange(9).reshape((3, 3))
print(x2)
print(x2.shape, x2.stride())
print("-----")

shape_env = ShapeEnv()
# t1 = torch.ones(2, 2, 768)
# t1 = torch.ones(2, 2, 768)
with FakeTensorMode(shape_env=shape_env) as fake_mode:
    t = fake_mode.from_tensor(
        x2,
        symbolic_context=StatelessSymbolicContext(
            dynamic_sizes=[
                DimDynamic.STATIC,
                DimDynamic.STATIC,
            ],
            dynamic_strides=[
                DimDynamic.DYNAMIC,
                DimDynamic.STATIC,
            ],
        ),
    )

    print(t, t.shape, t.stride())
~~~


Is this what you are hoping for?
~~~
FakeTensor(..., size=(3, 3), dtype=torch.int64) torch.Size([3, 3]) (s1, 1)
~~~

Reply:
I typoed, should have been static sizes lol